Backend/Core/Services/OrderService.cs

```diff
@@ -305,9 +326,45 @@ public async Task ConfirmDeliveryAsync(string userId, int orderId)
 
                 ExtensionMethods.OrderNullCheck(order);
 
-                order.IsAcceptedByCourier = true;
+                if (order.IsAcceptedByCourier)
+                {
+                    throw new HttpException(
+                        ErrorMessages.OrderAlreadyConfirmed,
+                        HttpStatusCode.InternalServerError);
+                }
 
-                await _orderRepository.UpdateAsync(order);
+                await ChangeIsAcceptedAsync(userRole, order, true);
+            }
+        }
+
+        public async Task RejectDeliveryConfirmationAsync(string userId, int orderId)
+        {
+            var user = await _userRepository.GetByIdAsync(userId);
+
+            ExtensionMethods.UserNullCheck(user);
+
+            var userRole = await _identityRoleService.GetUserRoleAsync(user);
+
+            if (userRole == IdentityRoleNames.User.ToString())
+            {
+                var order = await _orderRepository.SingleOrDefaultAsync(
+                    new OrderSpecification.GetByClientConfirmedDelivery(userId, orderId));
+
+                ExtensionMethods.OrderNotConfirmedClientCheck(order);
+                ExtensionMethods.OrderNullCheck(order);
+
+                await ChangeIsAcceptedAsync(userRole, order, false);
+
+            }
+            else if (userRole == IdentityRoleNames.Courier.ToString())
+            {
+                var order = await _orderRepository.SingleOrDefaultAsync(
+                    new OrderSpecification.GetByCourierConfirmedDelivery(userId, orderId));
+
+                ExtensionMethods.OrderNotConfirmedCourierCheck(order);
+                ExtensionMethods.OrderNullCheck(order);
```

Comment:
You get the confirmed order, and then check whether it isn't confirmed (the logic you wrote is impossible), throw an error